src/requests/parameter_wrappers.rs

```diff
@@ -87,6 +87,46 @@ fn cql_value_vec_from_tuple(
     Ok(res)
 }
 
+/// It requires that each element of the `arr` array is in [String, CqlValue] format.
+fn cql_value_vec_from_udt(
+    typ: &ComplexType,
+    arr: &Array,
+) -> napi::Result<Vec<(String, Option<CqlValue>)>> {
+    let mut res = vec![];
+    let support_types = typ.get_inner_types();
+    let parsing_error = || {
+        napi::Error::new(
+            Status::InvalidArg,
+            "Unexpected data when parsing parameters row".to_owned(),
+        )
+    };
+
+    if support_types.len() != arr.len().try_into().unwrap() {
+        return Err(napi::Error::new(
+            Status::InvalidArg,
+            "UDT has different amount of types and values".to_owned(),
+        ));
+    }
+
+    for (i, typ) in support_types.into_iter().enumerate() {
+        let pair = arr
+            .get::<Array>(i.try_into().unwrap())?
+            .ok_or_else(parsing_error)?;
+        if pair.len() != 2 {
+            return Err(parsing_error());
+        }
+        let name = pair.get::<String>(0)?.ok_or_else(parsing_error)?;
+
+        if let Ok(Some(_)) = pair.get::<Undefined>(1) {
+            res.push((name, None));
+        } else {
+            let value = cql_value_from_napi_value(&typ, &pair, 1)?;
+            res.push((name, Some(value)));
+        }
```

Comment:
:wrench: Please use `match` and explicit patterns.
Now, `Ok(None)` case leads to the `else` branch. Will it be handled correctly?